src/Helpers/OidStripper.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Helpers;
+
+use SimPod\PhpSnmp\Transport\SnmpClient;
+use function array_key_first;
+use function assert;
+use function is_string;
+use function Safe\substr;
+use function strlen;
+use function strrpos;
+
+final class OidStripper
+{
+    /**
+     * @param array<string, mixed> $leafOidData
+     *
+     * @return array<int, mixed>
+     *
+     * @template T
+     * @psalm-param array<string, T> $leafOidData
+     * @psalm-return array<int, T>
+     */
+    public static function stripParent(array $leafOidData) : array
+    {
+        $firstKey = array_key_first($leafOidData);
+        assert(is_string($firstKey));
+
+        $lastDotPos = strrpos($firstKey, '.');
+        assert($lastDotPos !== false);
```

Comment:
This should be a runtime check

Reply:
again, if user uses result from snmp client, the dot will always be there